src/views/ApartmentApp/components/StudentApplication/index.js

```diff
@@ -0,0 +1,884 @@
+      return <GordonLoader size={loaderSize} />;
+    }
+  } else {
+    return (
+      <Button
+        disabled={disabled || saving}
+        variant="contained"
+        color="primary"
+        fullWidth
+        onClick={onClick}
+      >
+        Save & Continue
+      </Button>
+    );
+  }
+};
+
+const StudentApplication = ({ userProfile, authentication }) => {
+  const [loading, setLoading] = useState(true);
+  const [saving, setSaving] = useState(false);
+  const [unsavedChanges, setUnsavedChanges] = useState(false);
+
+  const [applicationID, setApplicationID] = useState(-1); // Default value of -1 indicate to backend that the application ID number is not yet known
+  const [dateSubmitted, setDateSubmitted] = useState(null); // The date the application was submitted, or null if not yet submitted
+  const [dateModified, setDateModified] = useState(null); // The date the application was submitted, or null if not yet submitted
+  const [editorUsername, setEditorUsername] = useState(null); // The username of the application editor
+  const [applicants, setApplicants] = useState([]);
+  const [preferredHalls, setPreferredHalls] = useState([]); // Properties 'HallName' and 'HallRank' must be capitalized to match the backend
+
+  const [applicationCardsOpen, setApplicationCardsOpen] = useState(false);
+  const [newEditorProfile, setNewEditorProfile] = useState(null); // Stores the StudentProfileInfo of the new editor before the user confirms the change
+  const [changeEditorDialogOpen, setChangeEditorDialogOpen] = useState(false);
+  const [submitDialogOpen, setSubmitDialogOpen] = useState(false); // Use this for submitting app (later feature)
+  const [snackbarOpen, setSnackbarOpen] = useState(false);
+  const [snackbarText, setSnackbarText] = useState('');
+  const [snackbarSeverity, setSnackbarSeverity] = useState('');
+  const [saveButtonAlertTimeout, setSaveButtonAlertTimeout] = useState(null);
+
+  /**
+   * Loads the user's saved apartment application, if one exists
+   */
+  const loadSavedApplication = useCallback(async () => {
+    // TODO: Implement this once save/load of application data has been implemented in the backend
+    setLoading(true);
+    // Check if the current user is on an application. Returns the application ID number if found
+    let newApplicationID = null;
+    try {
+      newApplicationID = await housing.getCurrentApplicationID();
+      //! DEBUG
+      console.log('Retrieved Application ID: ' + newApplicationID);
+      if (newApplicationID === null || newApplicationID === -1) {
+        // Intentionally trigger the 'catch'
+        throw new Error("Invalid value of 'newApplicationID' = " + newApplicationID);
+      }
+      setApplicationID(newApplicationID);
+      let applicationDetails = await housing.getApartmentApplication(newApplicationID);
+      if (applicationDetails) {
+        if (applicationDetails.DateSubmitted) {
+          setDateSubmitted(applicationDetails.DateSubmitted);
+        }
```

Comment:
I believe you can use JavaScript's nullish coalescing to drastically simplify this entire `if` block. For example:
```suggestion
  setDateSubmitted(applicationDetails?.DateSubmitted ?? null);
```

This should work for all of the `set...`'s below.